eng/targetingpacks.targets

```diff
@@ -188,4 +188,13 @@
                      DotNetHostPath="$(DotNetTool)" />
     </ItemGroup>
   </Target>
+
+  <Target Name="UpdateRuntimePackToAddWasiRID" BeforeTargets="ProcessFrameworkReferences">
+    <!-- Always add wasi-wasm since it isn't added by sdk at all -->
+    <ItemGroup>
+      <KnownRuntimePack Update="$(LocalFrameworkOverrideName)"
+        RuntimePackRuntimeIdentifiers="%(RuntimePackRuntimeIdentifiers);wasi-wasm"
+        Condition="'$(UseLocalTargetingRuntimePack)' == 'true' and '%(RuntimePackLabels)' == 'Mono' and $([MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', '8.0'))" />
+    </ItemGroup>
+  </Target>
```

Comment:
Are you sure this is necessary?

wasi-wasm is already added to the live runtime pack: https://github.com/dotnet/runtime/blob/a54adf7f51b32d283bdd6f11767c065ed8c9a1e4/eng/targetingpacks.targets#L56C102-L56C102

Reply:
IIUC, that is added only if the sdk didn't already have an entry for 9.0 . Because we are now using a sdk that has it, that line won't add anything. And the sdk does not itself add `wasi-wasm`, so we need to always add it unconditionally.

Reply:
Oh yes that makes sense. We should move this out of the target and make it use the same Update syntax as https://github.com/dotnet/runtime/blob/65fbeb45ab70ca62e5d161bdddf0bf9022f12543/eng/targetingpacks.targets#L51. I think it should be placed in the same ItemGroup just with a different condition.

`<KnownRuntimePack Update="@(KnownRuntimePack->WithMetadataValue('Identity', '$(LocalFrameworkOverrideName)')->WithMetadataValue('RuntimeBackLabels', 'Mono')` and remove the wasi-wasm entry from above so that we only set it once.

Reply:
Outside the target I can do:
```xml
    <KnownRuntimePack Update="@(KnownRuntimePack->WithMetadataValue('Identity', '$(LocalFrameworkOverrideName)')->WithMetadataValue('RuntimePackLabels', 'Mono'))"
                      RuntimePackRuntimeIdentifiers="%(RuntimePackRuntimeIdentifiers);wasi-wasm"
                      Condition="'$(UseLocalTargetingRuntimePack)' == 'true'" />
```
.. but this should be done only for the runtime packs tfm>=8.0 . I can't use batching here, so I'm not sure how to restrict this with a condition `$([MSBuild]::VersionGreaterThanOrEquals('....', '8.0'))` to check against `KnownRuntimePack.TargetFramework`. Any suggestions?

Reply:
Do you need both 8.0 and 9.0? I thought we are already consistently targeting net9.0 everywhere for our tests. You could have two of these update entries, one for NetCoreAppCurrent and one for net8.0.

Asking as `UseLocalTargetingRuntimePack` suggests that we use the live runtime and targeting pack which is `net9.0`.

Reply:
Good point. Only `net9.0` is required.